dso-l2/src/main/java/com/tc/services/PlatformServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.tc.services;
+
+import com.tc.handler.CallbackDumpHandler;
+import com.tc.objectserver.impl.DistributedObjectServer;
+import com.tc.objectserver.impl.ServerBuilder;
+import org.terracotta.monitoring.PlatformService;
+
+/**
+ * @author vmad
+ */
+public class PlatformServiceImpl implements PlatformService {
+
+    private final DistributedObjectServer.DumpHelper dumpHelper;
+
+    public PlatformServiceImpl(DistributedObjectServer.DumpHelper dumpHelper) {
```

Comment:
If this type is being referenced from within the DistributedObjectServer, you probably want to float it out into its own file.

This kind of approach typically makes sense if the inner class isn't referenced from outside (it is just the implementation of an interface, for example) or if it is something directly associated with the class it is inside (an interface required to interact with it, an enum defining its constants, etc).

In this case, it is more like a cyclic dependency so it should probably be split out.